### DISCUSSION TITLE
[v5] Delete Many-To-Many relationship from a parent

### DISCUSSION THREAD

**Discussion by melokki** [adonisjs/core#1759] [Help] [answered]
Hello :wave: :upside_down_face: ,
I am trying to delete a many-to-many relationship from a parent relationship, but the entries from the pivot table are not deleted

this is what I have so far
```js
const course = await Course.findOrFail(params.id);
course.preload("subscriptions", (query) => {
    query.delete();
});
await course.delete();
```
My relationships are as follows:
```sql
Course -> hasMany -> Subscription
Subscription -> Many-To-Many -> User
```

Don't know how to gain access to that `detach` method :thinking:

**Comment by thetutlage** [comment]
The `preload` method is used to select the related rows and hence cannot be used to delete rows. You can access the detach method as follows:

```ts
course.related('subscriptions').detach()
```

It's cover in documentation too https://preview.adonisjs.com/guides/model-relations/many-to-many#detach

**Reply by melokki** [reply]
thank you for your reply

Yes, but my many-to-many relationship is not on `course -> subscriptions`, it's on `subscriptions -> users` so `course.related('subscriptions').detach()` won't work
```
Property 'detach' does not exist on type 'HasManyClientContract<HasManyRelationContract
```

**Comment by melokki** [comment, marked as answer]
I came up with something like this after all
```ts
await course.preload("subscriptions");
course.subscriptions.map(async (subscription) => {
    await subscription.related("users").detach();
    await subscription.delete();
});
```

I was trying to avoid that `.map()` but it appears it's not possible
